crypto/x509/x_x509.c

```diff
@@ -246,13 +252,14 @@ int X509_get_signature_nid(const X509 *x)
 }
 
 #ifndef OPENSSL_NO_SM2
-void X509_set_sm2_id(X509 *x, ASN1_OCTET_STRING *sm2_id)
+void X509_set0_sm2_id(X509 *x, ASN1_OCTET_STRING *sm2_id)
```

Comment:
There should be a test for this function.

Reply:
This is function is called in the `verify` app which is tested already, so do we need to add another C code test case to test this function separately?

Reply:
IMO, all newly introduced public API functions should have an associated test - it's not sufficient to test them "through" the command line apps. For setters it can be as simple as "call the set function and then check that a subsequent get returns the same value".

Reply:
Okay, I will add new test for this API